run-tests.py

```diff
@@ -78,6 +103,26 @@ def cleanup(out):
          'pr57580.c',
          ]
 
+
+def run(compiler_executable: str, compiler_args: List[str]) -> Tuple[int, str, str]:
+  """Execute a compiler command and capture its exit code, stdout, and stderr."""
+  cmd = [compiler_executable, *compiler_args]
+
+  try:
+    with subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True, encoding="utf-8") as process:
+      stdout, stderr = process.communicate()
+      exit_code = process.returncode
+  except FileNotFoundError as e:
+    # Compiler not found
+    return (127, "", f"{e}")
+  except Exception as e:
+    return (1, "", f"{e}")
```

Comment:
This simulates the exitcodes the shell might generate which might be misleading - we should specify our own so they can be differentiated.